src/register/register.ts

```diff
@@ -146,10 +316,49 @@ export class Register {
         })
       );
 
-      Register.registers[register].text = text;
-    }
+      // Harmonize newline character
+      text = text.replace(/\r\n/g, '\n');
```

Comment:
The following should work:

`let eol = vscode.workspace.getConfiguration('files').get("eol", "\n");`

and then you can add this as a new member to `configuration.ts` to expose it to your current functionality.